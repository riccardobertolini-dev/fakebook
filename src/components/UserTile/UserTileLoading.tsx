import { ImageContainer, UserInfo, LikeButton, QuoteLoading, SocialInteractions, SocialWallTile, EmptyProfileImage } from './UserTileLoading.styles';

type LoadingProps = {
	times: number;
};

export const UserTileLoading = ({ times }: LoadingProps) => {
	return (
		<>
			{[...Array(times)].map(() => (
				<SocialWallTile>
					<UserInfo>
						<EmptyProfileImage />
						<span></span> <span></span>
						<span></span>
					</UserInfo>
					<QuoteLoading></QuoteLoading>
					<ImageContainer></ImageContainer>
					<SocialInteractions>
						<LikeButton></LikeButton>
						<LikeButton></LikeButton>
					</SocialInteractions>
				</SocialWallTile>
			))}
		</>
	);
};
