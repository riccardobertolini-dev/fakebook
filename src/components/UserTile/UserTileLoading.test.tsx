import * as React from 'react';
import renderer from 'react-test-renderer';
import { UserTileLoading } from './UserTileLoading';

jest.mock('react-router-dom', () => ({
	Routes: jest.fn(),
	Route: jest.fn(),
	Link: jest.fn(),
}));

describe('UserTileLoading component', () => {
	it('should renders', () => {
		const wrapper = renderer.create(<UserTileLoading times={1} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
