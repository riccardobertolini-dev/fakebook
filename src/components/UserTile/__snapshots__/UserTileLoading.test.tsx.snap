// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTileLoading component should renders 1`] = `
<div
  className="sc-bcXHqe kgBFga"
>
  <div
    className="sc-gswNZR hTTcLS"
  >
    <div
      className="sc-hLBbgP jlFNFl"
    />
    <span />
     
    <span />
    <span />
  </div>
  <div
    className="sc-eDvSVe EJluP"
  />
  <div
    className="sc-dkrFOg cvLPLY"
  />
  <div
    className="sc-jrcTuL hFXkiT"
  >
    <a
      className="sc-kDvujY bkYPUV"
    />
    <a
      className="sc-kDvujY bkYPUV"
    />
  </div>
</div>
`;
